### PR TITLE
Feature/add jenkins integration

### PR DESCRIPTION
**Pull request by HenriChabert**

### CODE REVIEW

File: tests/python/integration/test_scenario.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+def test_extract_recipe(user_dss_clients):
+    add_integration_test(user_dss_clients, "Recipe_-_Extract_features")
+
+
+def test_score_recipe(user_dss_clients):
+    add_integration_test(user_dss_clients, "Recipe_-_Classify_images")
+
+
+def test_retrain_recipe(user_dss_clients):
```
Comment (MayeulRousselet) [inline tests/python/integration/test_scenario.py:23]: - Same as above, print the missing files from the Filenames check

File: tests/python/integration/test_scenario.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+def test_advanced_retrain(user_dss_clients):
+    add_integration_test(user_dss_clients, "Recipe_-_Advanced_Retrain")
+
+
+def test_model_download(user_dss_clients):
+    add_integration_test(user_dss_clients, "Macro_-_Model_Download")
+
+
+def test_cloud_integration(user_dss_clients):
```
Comment (MayeulRousselet) [inline tests/python/integration/test_scenario.py:35]: - The scenario runs check on  resnet_retrained_with_aws_images (step#3), but no check is defined.

**Review by MayeulRousselet — CHANGES_REQUESTED** [review]
Very good work Mr Henri.
What can be improved:
- I would rather prefer one scneario per model. Overkill? No! Because if we have an issue with a model it will be way easier to spot.
- The tests case (scenarii) can be quite complex. It would help your fellow devs if you can comment them in tests/python/integration/test_scenario.py. Explain what the scenario does and what it will check.
- It lacks a test in a container. Not trivial to set up but mandatory for this plugin as it is a prerequisite of this new version. Maybe in a v2.1 version?

File: python-lib/dku_deeplearning_image/utils.py [hunk 3]
```diff
@@ -230,6 +233,10 @@ def dbg_msg(msg, title=''):
 def log_info(*args):
     logger.info(*args)
 
+
+def log_warning(*args):
```
[the remark this answers is not in the document]
Comment (HenriChabert) [reply python-lib/dku_deeplearning_image/utils.py:237]: Actually, I think it is helpful as thanks to that, I only have to define in one place the logger (in utils) rather that in each file I use `logger.info` or `logger.warn`. What do you think?

File: Makefile [hunk 4]
```diff
@@ -1,10 +1,57 @@
...
+	@cat plugin.json | json_pp > /dev/null
+	@rm -rf dist
+	@mkdir dist
+	@echo "{\"remote_url\":\"${remote_url}\",\"last_commit_id\":\"${last_commit_id}\"}" > release_info.json
+	@git archive -v -9 --format zip -o dist/${archive_file_name} HEAD
+	@zip -u dist/${archive_file_name} release_info.json
+	@rm release_info.json
+	@echo "[SUCCESS] Archiving plugin to dist/ folder: Done!"
+
+unit-tests:
```
[the remark this answers is not in the document]
Comment (HenriChabert) [reply Makefile:20]: I'd rather keep it as I might do some unit tests in next PR

File: tests/python/unit/requirements.txt [hunk 5]
```diff
@@ -0,0 +1 @@
+pytest==6.2.1
```
[the remark this answers is not in the document]
Comment (HenriChabert) [reply tests/python/unit/requirements.txt]: Fixed in 3619d94

File: tests/python/unit/test_dummy_module.py [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
```
[the remark this answers is not in the document]
Comment (HenriChabert) [reply tests/python/unit/test_dummy_module.py]: Fixed in 3619d94

File: Jenkinsfile [hunk 7]
```diff
@@ -0,0 +1,63 @@
+pipeline {
+   options { disableConcurrentBuilds() }
+   agent { label 'dss-plugin-tests'}
+   environment {
+        PLUGIN_INTEGRATION_TEST_INSTANCE="/home/jenkins-agent/instance_config.json"
+    }
+   stages {
+      stage('Run Unit Tests') {
```
[the remark this answers is not in the document]
Comment (HenriChabert) [reply Jenkinsfile]: Fixed in bf8b8a5

File: Makefile (hunk 4 above)
[the remark this answers is not in the document]
Comment (HenriChabert) [reply Makefile]: Fixed in bf8b8a5

File: tests/python/integration/test_scenario.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+def test_retrain_recipe(user_dss_clients):
+    add_integration_test(user_dss_clients, "Recipe_-_Retrain_model")
+
+
+def test_advanced_retrain(user_dss_clients):
+    add_integration_test(user_dss_clients, "Recipe_-_Advanced_Retrain")
+
+
+def test_model_download(user_dss_clients):
+    add_integration_test(user_dss_clients, "Macro_-_Model_Download")
```
[the remark this answers is not in the document]
Comment (HenriChabert) [reply tests/python/integration/test_scenario.py:32]: True, I'll fix that in test project

File: tests/python/integration/test_scenario.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+def test_cloud_integration(user_dss_clients):
+    add_integration_test(user_dss_clients, "Recipe_-_Cloud_Integration")
+
+
+def test_edge_cases(user_dss_clients):
+    add_integration_test(user_dss_clients, "Recipe_-_Edge_cases")
+
+
+def test_api_endpoint_deployment(user_dss_clients):
```
[the remark this answers is not in the document]
Comment (HenriChabert) [reply tests/python/integration/test_scenario.py:43]: True, I'll fix that in test project

File: tests/python/integration/test_scenario.py [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+def test_score_recipe(user_dss_clients):
+    add_integration_test(user_dss_clients, "Recipe_-_Classify_images")
+
+
+def test_retrain_recipe(user_dss_clients):
+    add_integration_test(user_dss_clients, "Recipe_-_Retrain_model")
+
+
+def test_advanced_retrain(user_dss_clients):
```
[the remark this answers is not in the document]
Comment (HenriChabert) [reply tests/python/integration/test_scenario.py:27]: True, I'll fix that in test project

File: tests/python/integration/test_scenario.py [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+from dku_plugin_test_utils import dss_scenario
+
+TEST_PROJECT_KEY = "PLUGINTESTDEEPLEARNINGIMAGE"
+
+
+def add_integration_test(user_dss_clients, scenario_id):
+    dss_scenario.run(user_dss_clients, project_key=TEST_PROJECT_KEY, scenario_id=scenario_id)
+
+
+def test_run_tensorboard_start(user_dss_clients):
```
[the remark this answers is not in the document]
Comment (HenriChabert) [reply tests/python/integration/test_scenario.py:11]: True, I'll fix that in test project

File: tests/python/integration/test_scenario.py [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+    add_integration_test(user_dss_clients, "Webapp_-_Tensorboard")
+
+
+def test_extract_recipe(user_dss_clients):
```
[the remark this answers is not in the document]
Comment (HenriChabert) [reply tests/python/integration/test_scenario.py]: Fixed in 6fdc96e

File: tests/python/integration/test_scenario.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (HenriChabert) [reply tests/python/integration/test_scenario.py:32]: Done in my test project

File: tests/python/integration/test_scenario.py [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+    add_integration_test(user_dss_clients, "Recipe_-_Extract_features")
+
+
+def test_score_recipe(user_dss_clients):
```
[the remark this answers is not in the document]
Comment (HenriChabert) [reply tests/python/integration/test_scenario.py]: Did so in my test project.

File: tests/python/integration/test_scenario.py (hunk 1 above)
Comment (HenriChabert) [reply tests/python/integration/test_scenario.py]: Did so in my test project

File: tests/python/integration/test_scenario.py (hunk 2 above)
Comment (HenriChabert) [reply tests/python/integration/test_scenario.py:35]: True! Changed in my test project

File: tests/python/integration/test_scenario.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (HenriChabert) [reply tests/python/integration/test_scenario.py:27]: Changed

File: tests/python/integration/test_scenario.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (HenriChabert) [reply tests/python/integration/test_scenario.py]: As it is not possible, won't do

File: python-lib/dku_deeplearning_image/utils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (HenriChabert) [reply python-lib/dku_deeplearning_image/utils.py]: I resolve as it is minor :)

[the remark this answers is not in the document]
Comment (MayeulRousselet) [reply python-lib/dku_deeplearning_image/utils.py]: ok, so logger = logging.getLogger(__name__) will alway resolve to logging.getLogger("utils")?

[the remark this answers is not in the document]
Comment (HenriChabert) [reply python-lib/dku_deeplearning_image/utils.py]: Yep, isn't a good thing?

[the remark this answers is not in the document]
Comment (MayeulRousselet) [reply python-lib/dku_deeplearning_image/utils.py]: it depends if you want to have the precise module where the logging was called or if it is not important

[the remark this answers is not in the document]
Comment (HenriChabert) [reply python-lib/dku_deeplearning_image/utils.py]: You are right, it is not ideal at all... I will change that

[the remark this answers is not in the document]
Comment (HenriChabert) [reply python-lib/dku_deeplearning_image/utils.py]: Did it in 59e2a3

**Review by MayeulRousselet — APPROVED** [review]
LGMT

**Pull request merged**